pkg/common/cns-lib/volume/manager.go

```diff
@@ -563,6 +566,19 @@ func (m *defaultManager) createVolumeWithImprovedIdempotency(ctx context.Context
 	return resp, faultType, err
 }
 
+// IsTaskPending returns true in two cases -
+// 1. if the task status was in progress
+// 2. if the status was an error but the error was for adding the task to the listview (as we don't know the status of the task on CNS)
+func IsTaskPending(volumeOperationDetails *cnsvolumeoperationrequest.VolumeOperationRequestDetails) bool {
+	if volumeOperationDetails.OperationDetails.TaskStatus == taskInvocationStatusInProgress &&
+		volumeOperationDetails.OperationDetails.TaskID != "" {
+		return true
+	} else if volumeOperationDetails.OperationDetails.TaskStatus == taskInvocationStatusError && strings.Contains(volumeOperationDetails.OperationDetails.Error, listviewAdditionError) {
```

Comment:
This is not ideal. A custom error type doesn't help here as we are storing the task error as a `string` in the `cnsvolumeoperationrequest` custom resource.